internal/exec/stages/files/files.go

```diff
@@ -76,5 +85,87 @@ func (s stage) Run(config types.Config) bool {
 		return false
 	}
 
+	// add systemd unit to relabel files
+	if err := s.addRelabelUnit(config); err != nil {
+		s.Logger.Crit("failed to add relabel unit: %v", err)
+		return false
+	}
+
 	return true
 }
+
+// do some upfront work to find out if it's even worth collecting filenames
+func (s *stage) checkRelabeling() error {
+	if distro.RestoreconCmd() == "" {
+		s.Logger.Debug("compiled without restorecon, skipping relabel")
+		return nil
+	}
+
+	_, err := os.Stat(s.JoinPath(distro.RestoreconCmd()))
+	switch {
+	case os.IsNotExist(err):
+		s.Logger.Debug("targeting root without %s, skipping relabel", distro.RestoreconCmd())
+		return nil
+	case err != nil:
+		return err
+	}
+
+	v := make([]string, 0)
+	s.toRelabel = &v
+	return nil
+}
+
+func (s stage) relabel(paths ...string) {
+	if s.toRelabel != nil {
+		*s.toRelabel = append(*s.toRelabel, paths...)
+	}
+}
+
+func (s stage) addRelabelUnit(config types.Config) error {
+	if s.toRelabel == nil || len(*s.toRelabel) == 0 {
+		return nil
+	}
+
+	// create the unit file itself
+	unit := types.Unit{
+		Name: "ignition-relabel.service",
+		Contents:
+`[Unit]
+Description=Relabel ignition files
+DefaultDependencies=no
+After=local-fs.target
+Before=sysinit.target
+ConditionSecurity=selinux
```

Comment:
So this will still execute on CL...there aren't any plans to change to enforcing are there?  I'd say we make this a build-time option and disable it on CL - that's simple and safe.

Reply:
Right, it'll still execute in permissive mode. Which I think we want, right? Might as well relabel it if a policy is loaded, regardless of whether we're enforcing vs permissive.

I implemented @ajeddeloh's suggestion to just allow `-X .../restoreconCmd=""` for build-time disabling. I'm open to making a more explicit e.g. `useSelinux` var instead.